### PR TITLE
Expose docs to main website

### PR DESCRIPTION
**Pull request by mhadaily**
## Description

This PR will 

- add a placeholder for missing docs 
- add null safety announcement 
- add null safety page in docs
- expose docs on the main website
- upgrade node to 14.15.x and npm 7
- update packages
- update overview on the docs page

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [X] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: docs/overview.mdx [hunk 1]
```diff
@@ -6,33 +6,41 @@ hide_title: true
...
+| [Network Info Plus](battery/overview)          | <img src="https://img.shields.io/pub/v/network_info_plus.svg" />          |
+| [Package Info Plus](battery/overview)          | <img src="https://img.shields.io/pub/v/package_info_plus.svg" />          |
+| [Sensors Plus](battery/overview)               | <img src="https://img.shields.io/pub/v/sensors_plus.svg" />               |
+| [Share Plus](battery/overview)                 | <img src="https://img.shields.io/pub/v/share_plus.svg" />                 |
+| [Android Alarm Manager Plus](battery/overview) | <img src="https://img.shields.io/pub/v/android_alarm_manager_plus.svg" /> |
+| [Android Intent Plus](battery/overview)        | <img src="https://img.shields.io/pub/v/android_intent_plus.svg" />        |
+
+:::note
+PlusPlugins are all null safety compatible. [check this out](null_safety)
 :::
```
Comment (pr-Mais) [inline docs/overview.mdx:33]: we don't need the lines for installing battery_plus here

**Review by pr-Mais — APPROVED** [review]

File: docs/android_intent_plus/usage.mdx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+title: Android Intent Plus Plus Usage
```
Comment (jpnurmi) [inline docs/android_intent_plus/usage.mdx:2]: extra plus

File: docs/android_intent_plus/usage.mdx [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+sidebar_label: Usage
+hide_title: true
+---
+
+## Using Android Intent Plus Plus
```
Comment (jpnurmi) [inline docs/android_intent_plus/usage.mdx:7]: extra plus

File: docs/device_info_plus/overview.mdx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+title: Device Info Overview
```
Comment (jpnurmi) [inline docs/device_info_plus/overview.mdx:2]: missing plus

File: docs/network_info_plus/overview.mdx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+---
+title: Network Info Overview
```
Comment (jpnurmi) [inline docs/network_info_plus/overview.mdx:2]: missing plus

File: docs/overview.mdx [hunk 6]
```diff
@@ -6,33 +6,41 @@ hide_title: true
...
+| [Battery Plus](battery/overview)               | <img src="https://img.shields.io/pub/v/battery_plus.svg" />               |
+| [connectivity Plus](battery/overview)          | <img src="https://img.shields.io/pub/v/connectivity_plus.svg" />          |
+| [Device Info Plus](battery/overview)           | <img src="https://img.shields.io/pub/v/device_info_plus.svg" />           |
+| [Network Info Plus](battery/overview)          | <img src="https://img.shields.io/pub/v/network_info_plus.svg" />          |
+| [Package Info Plus](battery/overview)          | <img src="https://img.shields.io/pub/v/package_info_plus.svg" />          |
+| [Sensors Plus](battery/overview)               | <img src="https://img.shields.io/pub/v/sensors_plus.svg" />               |
+| [Share Plus](battery/overview)                 | <img src="https://img.shields.io/pub/v/share_plus.svg" />                 |
+| [Android Alarm Manager Plus](battery/overview) | <img src="https://img.shields.io/pub/v/android_alarm_manager_plus.svg" /> |
+| [Android Intent Plus](battery/overview)        | <img src="https://img.shields.io/pub/v/android_intent_plus.svg" />        |
```
Comment (jpnurmi) [inline docs/overview.mdx:29]: the links are all pointing to battery/overview

File: docs/package_info_plus/overview.mdx [hunk 7]
```diff
@@ -0,0 +1,9 @@
+---
+title: Package Info Overview
```
Comment (jpnurmi) [inline docs/package_info_plus/overview.mdx:2]: missing plus

File: docs/share_plus/usage.mdx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+---
+title: Network Plus Usage
```
Comment (jpnurmi) [inline docs/share_plus/usage.mdx:2]: s/network/share/

File: docs/share_plus/usage.mdx [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+sidebar_label: Usage
+hide_title: true
+---
+
+## Using Network Info Plus
```
Comment (jpnurmi) [inline docs/share_plus/usage.mdx:7]: s/network/share/

File: docs/sidebars.js [hunk 10]
```diff
@@ -8,8 +8,51 @@ function toReferenceAPI(plugin) {
...
+      "network_info_plus/overview",
+      "network_info_plus/usage",
+      toReferenceAPI("network_info"),
+    ],
+    ["Package Info"]: [
+      "package_info_plus/overview",
+      "package_info_plus/usage",
+      toReferenceAPI("package_info"),
+    ],
+    ["Device Info"]: [
```
Comment (jpnurmi) [inline docs/sidebars.js:32]: could move this above network info for alphabetical order?

**Comment by jpnurmi** [comment]
in `null_safety.mdx`

> Not all PlusPlugins packages have null-safety versions available yet. If you need to use the null-safe packages alongside the non-null-safe packages then it's recommended you switch to sourcing your PlusPlugins packages directly from the main PlusPlugins repository (rather than pub.dev) where the packages have compatible version constraints.

This can be removed now that we have them all migrated

**Comment by jpnurmi** [comment]
Awesome! :+1:

File: docs/overview.mdx (hunk 1 above)
Comment (mhadaily) [reply docs/overview.mdx:33]: Now I added all of them just for example.

File: docs/overview.mdx (hunk 6 above)
Comment (mhadaily) [reply docs/overview.mdx:29]: Good catch

**Review by jpnurmi — APPROVED** [review]
Very nice! LGTM :+1:

**Pull request merged**